# **Business Understanding:**

Analyze  the Calls made in the company and try and finc out areas of improvement

**Data:**

[Data Source](https://www.kaggle.com/datasets/akash1vishwakarma/call-center-dataset)

#### **Load Libraries**

In [1]:
import matplotlib.pyplot as plt 
import pandas as pd 
import numpy as np 
import seaborn as sns 
import scipy as stats

#### **Load Data**

In [2]:
df = pd.read_excel('data/01 Call-Center-Dataset.xlsx')

# **Data Understanding:**
- glimpse of the data(head, tail)
- shape(dimensions)
- columns
- information
- data type
- duplicates
- outliers
- null values
- descriptive stats (quantitative and qualitative)

#### **Glimpse of the data:**

In [3]:
df.head()

,Call Id,Agent,Date,Time,Topic,Answered (Y/N),Resolved,Speed of answer in seconds,AvgTalkDuration,Satisfaction rating
0,ID0001,Diane,2021-01-01,09:12:58,Contract related,Y,Y,109.0,00:02:23,3.0
1,ID0002,Becky,2021-01-01,09:12:58,Technical Support,Y,N,70.0,00:04:02,3.0
2,ID0003,Stewart,2021-01-01,09:47:31,Contract related,Y,Y,10.0,00:02:11,3.0
3,ID0004,Greg,2021-01-01,09:47:31,Contract related,Y,Y,53.0,00:00:37,2.0
4,ID0005,Becky,2021-01-01,10:00:29,Payment related,Y,Y,95.0,00:01:00,3.0


In [4]:
df.tail()

,Call Id,Agent,Date,Time,Topic,Answered (Y/N),Resolved,Speed of answer in seconds,AvgTalkDuration,Satisfaction rating
4995,ID4996,Jim,2021-03-31,16:37:55,Payment related,Y,Y,22.0,00:05:40,1.0
4996,ID4997,Diane,2021-03-31,16:45:07,Payment related,Y,Y,100.0,00:03:16,3.0
4997,ID4998,Diane,2021-03-31,16:53:46,Payment related,Y,Y,84.0,00:01:49,4.0
4998,ID4999,Jim,2021-03-31,17:02:24,Streaming,Y,Y,98.0,00:00:58,5.0
4999,ID5000,Diane,2021-03-31,17:39:50,Contract related,N,N,NaN,NaN,NaN


#### **Dimensions, columns & datatypes**

In [5]:
rows, columns = df.shape
output = f"Records: {rows} and Features: {columns}"
print(output)

Records: 5000 and Features: 10


In [6]:
df.columns

Index(['Call Id', 'Agent', 'Date', 'Time', 'Topic', 'Answered (Y/N)',
       'Resolved', 'Speed of answer in seconds', 'AvgTalkDuration',
       'Satisfaction rating'],
      dtype='str')

In [8]:
df.dtypes

Call Id                           str
Agent                             str
Date                              str
Time                           object
Topic                             str
Answered (Y/N)                    str
Resolved                          str
Speed of answer in seconds    float64
AvgTalkDuration                object
Satisfaction rating           float64
dtype: object

#### **Info, duplicates, null values**

In [9]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 5000 entries, 0 to 4999
Data columns (total 10 columns):
 #   Column                      Non-Null Count  Dtype  
---  ------                      --------------  -----  
 0   Call Id                     5000 non-null   str    
 1   Agent                       5000 non-null   str    
 2   Date                        5000 non-null   str    
 3   Time                        5000 non-null   object 
 4   Topic                       5000 non-null   str    
 5   Answered (Y/N)              5000 non-null   str    
 6   Resolved                    5000 non-null   str    
 7   Speed of answer in seconds  4054 non-null   float64
 8   AvgTalkDuration             4054 non-null   object 
 9   Satisfaction rating         4054 non-null   float64
dtypes: float64(2), object(2), str(6)
memory usage: 390.8+ KB


In [10]:
df.duplicated().sum()

np.int64(0)

In [13]:
df.isna().sum()

Call Id                         0
Agent                           0
Date                            0
Time                            0
Topic                           0
Answered (Y/N)                  0
Resolved                        0
Speed of answer in seconds    946
AvgTalkDuration               946
Satisfaction rating           946
dtype: int64

In [12]:
df.isnull().sum() * 100 / len(df)


Call Id                        0.00
Agent                          0.00
Date                           0.00
Time                           0.00
Topic                          0.00
Answered (Y/N)                 0.00
Resolved                       0.00
Speed of answer in seconds    18.92
AvgTalkDuration               18.92
Satisfaction rating           18.92
dtype: float64

In [14]:
df = df.dropna(axis=0)

In [15]:
df.isnull().sum() * 100 / len(df)

Call Id                       0.0
Agent                         0.0
Date                          0.0
Time                          0.0
Topic                         0.0
Answered (Y/N)                0.0
Resolved                      0.0
Speed of answer in seconds    0.0
AvgTalkDuration               0.0
Satisfaction rating           0.0
dtype: float64

In [16]:
df.info()

<class 'pandas.DataFrame'>
Index: 4054 entries, 0 to 4998
Data columns (total 10 columns):
 #   Column                      Non-Null Count  Dtype  
---  ------                      --------------  -----  
 0   Call Id                     4054 non-null   str    
 1   Agent                       4054 non-null   str    
 2   Date                        4054 non-null   str    
 3   Time                        4054 non-null   object 
 4   Topic                       4054 non-null   str    
 5   Answered (Y/N)              4054 non-null   str    
 6   Resolved                    4054 non-null   str    
 7   Speed of answer in seconds  4054 non-null   float64
 8   AvgTalkDuration             4054 non-null   object 
 9   Satisfaction rating         4054 non-null   float64
dtypes: float64(2), object(2), str(6)
memory usage: 348.4+ KB


#### **Descriptive Stats(quantitative an dqualitative)**

In [17]:
df.describe()

,Speed of answer in seconds,Satisfaction rating
count,4054.000000,4054.000000
mean,67.520720,3.403552
std,33.592872,1.212220
min,10.000000,1.000000
25%,39.000000,3.000000
50%,68.000000,3.000000
75%,97.000000,4.000000
max,125.000000,5.000000


In [18]:
df.describe(include='object') # qualitative data

/tmp/ipykernel_12407/3942173830.py:1: Pandas4Warning: For backward compatibility, 'str' dtypes are included by select_dtypes when 'object' dtype is specified. This behavior is deprecated and will be removed in a future version. Explicitly pass 'str' to `include` to select them, or to `exclude` to remove them and silence this warning.
See https://pandas.pydata.org/docs/user_guide/migration-3-strings.html#string-migration-select-dtypes for details on how to write code that works with pandas 2 and 3.
  df.describe(include='object') # qualitative data


,Call Id,Agent,Date,Time,Topic,Answered (Y/N),Resolved,AvgTalkDuration
count,4054,4054,4054,4054,4054,4054,4054,4054
unique,4054,8,90,375,5,1,2,391
top,ID0001,Jim,2021-01-11,11:55:41,Streaming,Y,Y,00:04:43
freq,1,536,68,27,847,4054,3646,22


In [19]:
(3646/4054)*100

89.93586581154416

In [20]:
(536/4054)*100


13.22150962012827

Insights: 
- we have missing values in the  Speed of answer in seconds, AvgTalkDuration and Satisfaction rating  features
- no duplicates found
- we had 19% of missing values in the dataset the **remedy** was to drop it as the count was small

**Speed of answer in seconds:**
- It took an average of 67.52s to answer the calls 
- The calls were spread out at approx.  33.59sec from the (arithmetic) average response time
- The fastest reponse time was at 10sec
- The longest wait time to respond to a call was 125sec
- 25% of the calls took less than 39 sec of response time
- 50% (median) of the call attendants took 68sec to answer the call
- 75% of the calls tok less than 97sec to be answered

**Agent:**
- Jim handled approximately 13% (536 calls) of the calls 

- approximately 90% of the issues were resolved
- we recive 100% calls answered
